src/Symfony/Component/Serializer/Deserialize/Template/LazyTemplateGenerator.php

```diff
@@ -0,0 +1,309 @@
+                                ]),
+                            ),
+                            new VariableNode('k'),
+                        )),
+                    ]),
+                ], new ArgumentsNode([
+                    new VariableNode('config'),
+                    new VariableNode('instantiator'),
+                    new VariableNode('providers', byReference: true),
+                ])),
+            )),
+        ];
+
+        $iterableValueNode = new FunctionCallNode(
+            new VariableNode('iterable'),
+            new ArgumentsNode([new VariableNode('resource'), new VariableNode('boundaries')]),
+        );
+
+        $returnNodes = [
+            new ExpressionNode(new ReturnNode(
+                'array' === $node->type->name() ? new FunctionCallNode('\iterator_to_array', new ArgumentsNode([$iterableValueNode])) : $iterableValueNode,
+            )),
+        ];
+
+        return [
+            new ExpressionNode(new AssignNode(
+                new ArrayAccessNode(new VariableNode('providers'), new PhpScalarNode($node->identifier())),
+                new ClosureNode(
+                    new ParametersNode(['resource' => 'mixed', 'offset' => 'int', 'length' => 'int']),
+                    ($node->type->isNullable() ? '?' : '').$node->type->name(),
+                    true,
+                    [...$getBoundariesNodes, ...$iterableClosureNodes, ...$returnNodes],
+                    new ArgumentsNode([
+                        new VariableNode('config'),
+                        new VariableNode('instantiator'),
+                        new VariableNode('providers', byReference: true),
+                    ]),
+                ),
+            )),
+            ...$this->providerNodes($node->item, $context),
+        ];
+    }
+
+    /**
+     * @param array<string, mixed> $context
+     *
+     * @return list<PhpNodeInterface>
+     */
+    protected function objectNodes(ObjectNode $node, array &$context): array
+    {
+        $getBoundariesNodes = [
+            new ExpressionNode(new AssignNode(new VariableNode('boundaries'), new MethodCallNode(
+                new PhpScalarNode('\\'.$this->splitterClassName),
+                'splitDict',
+                new ArgumentsNode([new VariableNode('resource'), new VariableNode('offset'), new VariableNode('length')]),
+                static: true,
+            ))),
+        ];
+
+        if ($node->type->isNullable()) {
+            $getBoundariesNodes[] = new IfNode(new BinaryNode('===', new PhpScalarNode(null), new VariableNode('boundaries')), [
+                new ExpressionNode(new ReturnNode(new PhpScalarNode(null))),
+            ]);
+        }
+
+        $propertyValueProvidersNodes = [];
+        $propertiesClosuresNodes = [];
+
+        foreach ($node->properties as $serializedName => $property) {
+            array_push($propertyValueProvidersNodes, ...$this->providerNodes($property['value'], $context));
```

Comment:
There will probably be a small performance gain by accumulating data in an array and calling `array_push` only one time after the loop.